src/PABLO/Octant.cpp

```diff
@@ -1584,6 +1584,205 @@ Octant Octant::computePeriodicOctant(uint8_t iface) const {
+ */
+Octant Octant::computeNodePeriodicOctant(uint8_t inode) const {
+    Octant degOct(this->m_dim, this->m_level, this->m_x, this->m_y, this->m_z);
+    uint32_t maxLength = uint32_t(1)<<TreeConstants::MAX_LEVEL;
+    uint32_t dh = this->getLogicalSize();
+
+    uint8_t iface1 = sm_treeConstants[m_dim].nodeFace[inode][0];
+    uint8_t iface2 = sm_treeConstants[m_dim].nodeFace[inode][1];
+    uint8_t iface3 = sm_treeConstants[m_dim].nodeFace[inode][m_dim-1];
+
+    if (!m_info[iface1] && !m_info[iface2] && !m_info[iface3]){
+        return *this;
+    }
+    else{
+        switch (iface1) {
+        case 0 :
+        {
+            if (m_info[0]){
+                degOct.m_x = maxLength-dh;
+            }
+            else{
+                degOct.m_x -= dh;
+            }
+        }
+        break;
+        case 1 :
+        {
+            if (m_info[1]){
+                degOct.m_x = 0;
+            }
+            else{
+                degOct.m_x += dh;
+            }
+        }
+        break;
+        }
+
+        switch (iface2) {
+        case 2 :
+        {
+            if (m_info[2]){
+                degOct.m_y = maxLength-dh;
+            }
+            else{
+                degOct.m_y -= dh;
+            }
+        }
+        break;
+        case 3 :
+        {
+            if (m_info[3]){
+                degOct.m_y = 0;
+            }
+            else{
+                degOct.m_y += dh;
+            }
+        }
+        break;
+        }
+
+        switch (iface3) {
+        case 4 :
+        {
+            if (m_info[4]){
+                degOct.m_z = maxLength-dh;
+            }
+            else{
+                degOct.m_z -= dh;
+            }
+        }
+        break;
+        case 5 :
+        {
+            if (m_info[5]){
+                degOct.m_z = 0;
+            }
+            else{
+                degOct.m_z += dh;
+            }
+        }
+        break;
+        }
+
+        degOct.m_level = this->m_level;
+        degOct.m_info = false;
+        return degOct;
+    }
+
+};
+
+/** Build a same size periodic octant of this octant throught edge inode.
+ * \return Periodic octant of the same size (note: it is a stand-alone octant,
+ * may be not living in octree).
+ * The faces through the octant is periodic has to be passed as boolean flags.
+ * \param[in] iedge Local index of the edge target.
+ */
+Octant Octant::computeEdgePeriodicOctant(uint8_t iedge) const {
+    Octant degOct(this->m_dim, this->m_level, this->m_x, this->m_y, this->m_z);
+    uint32_t maxLength = uint32_t(1)<<TreeConstants::MAX_LEVEL;
+    uint32_t dh = this->getLogicalSize();
+
+    std::array<uint8_t,2> iface;
+    iface[0] = sm_treeConstants[m_dim].edgeFace[iedge][0];
+    iface[1] = sm_treeConstants[m_dim].edgeFace[iedge][1];
+
+    int8_t          cxyz[3] = {0,0,0};
+    for (int idim=0; idim<m_dim; idim++){
+        cxyz[idim] = sm_treeConstants[m_dim].edgeCoeffs[iedge][idim];
+    }
+
```

Comment:
Can 'getEdgePeriodicCoord' be used to evaluate the coordinates instead of duplicating the code?

Reply:
No, they deal with periodic octant in different way.